repos/system_upgrade/el8toel9/actors/checkifcfg/libraries/ifcfg.py

```diff
@@ -0,0 +1,145 @@
+import os
+
+from leapp import reporting
+
+
+def process():
+    TRUE_VALUES = ['yes', 'true', '1']
+    GOOD_TYPES = [
+        'ethernet', 'ctc', 'wireless', 'infiniband', 'bond',
+        'team', 'vlan', 'bridge'
+    ]
+
+    rule_files = []
+    bad_type_files = []
+    not_controlled_files = []
+
+    try:
+        files = os.scandir('/etc/sysconfig/network-scripts')
+    except OSError:
+        # The legacy network-scripts package is probably already
+        # gone without a trace. Good!
+        return
+
+    for f in files:
+        bad_type = False
+        got_type = None
+        nm_controlled = True
+
+        if not f.is_file():
+            continue
+
+        if f.name.startswith('rule-') or f.name.startswith('rule6-'):
+            rule_files.append(f.path)
+            continue
+
+        if not f.name.startswith('ifcfg-'):
+            continue
+
+        if f.name == 'ifcfg-lo':
+            continue
+
+        for line in open(f.path):
+            try:
+                (key, value) = line.split('#')[0].strip().split('=')
+            except ValueError:
+                # We're not interested in lines that are not
+                # simple assignments. Play it safe.
+                continue
+
+            if key == 'TYPE':
+                if got_type is None:
+                    got_type = value.lower()
+                elif got_type != value.lower():
+                    bad_type = True
+
+            if key == 'BONDING_MASTER':
+                if got_type is None:
+                    got_type = 'bond'
+                elif got_type != 'bond':
+                    bad_type = True
+
+            if key == 'NM_CONTROLLED' and value.lower() not in TRUE_VALUES:
+                nm_controlled = False
+
+        if got_type not in GOOD_TYPES:
+            bad_type = True
+
+        # Don't bother reporting the file for NM_CONTROLLED=no
+        # if its type is not supportable with NetworkManager anyway
+        if bad_type is True:
+            bad_type_files.append(f.path)
+        elif nm_controlled is False:
+            not_controlled_files.append(f.path)
+
+    if rule_files:
+        title = 'Rule file detected'
```

Comment:
This is generic too much. From the title, I cannot even imagine what is it about. Imagine you see by default just this line. Create the title that gives you an idea about what the report is.

Reply:
(this is not review yet. just a one thing that I see right now)

Reply:
Tried to improve this in an updated patch. I'm not sure if it's good enough because I don't have a very clear idea about where does the user see this message. If it's still not clear enough, I'd be very thankful for suggestions.